src/GdbServer.cc

```diff
@@ -1741,7 +1794,49 @@ static ScopedFd generate_fake_proc_maps(Task* t) {
   return move(file.fd);
 }
 
-int GdbServer::open_file(Session& session, const std::string& file_name) {
+static bool is_ld_mapping(std::string map_name) {
+  char ld_start[] = "ld-";
+  size_t matchpos = map_name.find_last_of('/');
+  string fname = map_name.substr(matchpos == string::npos ? 0 : matchpos + 1);
+  return memcmp(fname.c_str(), ld_start,
+                sizeof(ld_start)-1) == 0;
+}
+
+static bool is_likely_interp(std::string fsname) {
+  return fsname == "/lib64/ld-linux-x86-64.so.2" || fsname == "/lib/ld-linux.so.2";
+}
+
+std::string normalize_path(std::string path)
```

Comment:
Please document what normalization is done.

Reply:
Can we merge this with `normalize_file_name` in `SourcesCommand.cc`? I apologise for not documenting that properly!